tests/test_generators.py

```diff
@@ -0,0 +1,329 @@
+                "name": "txt",
+                "chars": ["6", "2", "R", "W", "5", "3", "A", "4", "1", "L", "I", "O"],
+                "str_len_min": 2.0,
+                "str_len_max": 2.0,
+                "order": "random",
+            },
+            {
+                "generator": "integer",
+                "name": "int",
+                "min_value": 60661.0,
+                "max_value": 98168.0,
+                "order": "descending",
+            },
+            {
+                "generator": "float",
+                "name": "flo",
+                "min_value": 25.0392,
+                "max_value": 51.0,
+                "sig_figs": 6,
+                "order": "random",
+            },
+            {
+                "generator": "float",
+                "name": "flo",
+                "min_value": -122.9117,
+                "max_value": 127.02,
+                "sig_figs": 7,
+                "order": "random",
+            },
+            {
+                "generator": "integer",
+                "name": "int",
+                "min_value": 664.0,
+                "max_value": 9464.0,
+                "order": "random",
+            },
+        ]
+        generator.synthesize(20)
+
+        # cannot call this because the report will generate the char lists in different orders thus making this fail
+        # (even though the elements in the lists are the same)
+        # mock_generate_dataset.assert_called_once_with(generator.rng, expected_columns_to_generate, 20)
+
+        mock_generate_dataset.assert_called_once()
+        for i in range(len(expected_columns_to_generate)):
+            for key in generator.col_data[i].keys():
+                if isinstance(generator.col_data[i][key], list):
+                    self.assertTrue(
+                        set(generator.col_data[i][key]).issubset(
+                            expected_columns_to_generate[i][key]
+                        )
+                    )
+                else:
+                    self.assertEqual(
+                        generator.col_data[i][key], expected_columns_to_generate[i][key]
+                    )
+
+    def test_uncorrelated_synthesize_output(self):
+        data = dp.Data(os.path.join(test_dir, "data/tabular.csv"))
+        profile = dp.Profiler(
+            data=data,
+            options=self.profile_options,
+            samples_per_update=len(data),
+        )
+        generator = TabularGenerator(profile=profile, is_correlated=False, seed=42)
+        self.assertFalse(generator.is_correlated)
+        actual_synthetic_data = generator.synthesize(20)
+
+        self.assertEqual(len(actual_synthetic_data), 20)
+        self.assertIsInstance(actual_synthetic_data, pd.DataFrame)
+        np.testing.assert_array_equal(
+            actual_synthetic_data.columns.values,
+            np.array(["dat", "txt", "int", "flo"], dtype="object"),
+        )
+
+        # for column in actual_synthetic_data_columns:
```

Comment:
will delete all of this in a second, but essentially was trying to hardcode an expected, but obviously the data is generated randomly, so we can only check for general stuff: length, type, and column values of output